orca-bakery/src/main/groovy/com/netflix/spinnaker/orca/bakery/tasks/CreateBakeTask.groovy

```diff
@@ -188,10 +182,40 @@ class CreateBakeTask implements RetryableTask {
       requestMap.remove("baseName")
     }
 
+    if (stage.context.baseOs == null) {
```

Comment:
can you please also add unit tests for the new functionality?

Reply:
sure